api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -260,8 +260,24 @@ message Bootstrap {
   // This may be overridden on a per-cluster basis in cds_config, when
   // :ref:`dns_resolution_config <envoy_v3_api_field_config.cluster.v3.Cluster.dns_resolution_config>`
   // is specified.
+  // *dns_resolution_config* will be deprecated once
+  // :ref:'typed_dns_resolver_config <envoy_v3_api_field_config.bootstrap.v3.Bootstrap.typed_dns_resolver_config>'
+  // is fully supported.
   core.v3.DnsResolutionConfig dns_resolution_config = 30;
 
+  // DNS resolver type configuration extension. This extension can be used to configure c-ares, apple,
+  // or any other DNS resolver types and the related parameters. When c-ares DNS is configured,
+  // :ref:'dns_resolution_config <envoy_v3_api_field_config.bootstrap.v3.Bootstrap.dns_resolution_config>'
+  // will be enclosed in this *typed_dns_resolver_config*.
+  // This configuration is to replace the *dns_resolution_config* configuration eventually.
+  // During the transition period when both *dns_resolution_config* and *typed_dns_resolver_config* exists,
+  // this configuration is optional.
+  // When *typed_dns_resolver_config* is in place, Envoy will use it and ignore *dns_resolution_config*.
+  // when *typed_dns_resolver_config* is missing, the default behavior is in place. Envoy will either use
```

Comment:
nit: s/when/When here and in the other places.
BTW: do you mean the default behavior will be in effect when both `typed_dns_resolver_config` and `dns_resolution_config` are missing?

Reply:
That's right.